File: VB/WinFormsProject/ViewerForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.DashboardWin
Imports DevExpress.DashboardCommon
Imports DevExpress.DataAccess.ConnectionParameters

Namespace WinFormsProject
	Partial Public Class ViewerForm
		Inherits DevExpress.XtraEditors.XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Sub New(ByVal fileName As String)
			Me.New()
			dashboardViewer1.LoadDashboard(fileName)
		End Sub

		Private nwindFileName As String = "../../../nwind.mdb"
        Private Sub dashboardViewer1_ConfigureDataConnection(ByVal sender As Object, ByVal e As DashboardConfigureDataConnectionEventArgs) Handles dashboardViewer1.ConfigureDataConnection
            Select Case e.ConnectionName
                Case "nwindConnection"
                    Dim access As Access97ConnectionParameters = TryCast(e.ConnectionParameters, Access97ConnectionParameters)
                    If access IsNot Nothing Then
                        access.FileName = nwindFileName
                    End If
            End Select
        End Sub


		Private Sub UpdateDashboard(ByVal dashboard As Dashboard)
			dashboard.DataSources.Clear()
			Dim invoiceDataSource As New DataSource("RuntimeInvoices")
			dashboard.DataSources.Add(invoiceDataSource)
			For Each item In dashboard.Items.Where(Function(i) TryCast(i, DataDashboardItem) IsNot Nothing).Cast(Of DataDashboardItem)()
				item.DataSource = invoiceDataSource
			Next item
		End Sub

        Private Sub dashboardViewer1_DataLoading(ByVal sender As Object, ByVal e As DataLoadingEventArgs) Handles dashboardViewer1.DataLoading
            If e.DataSourceName = "RuntimeInvoices" Then
                e.Data = NwindDataAdapter.NwindHelper.GetInvoices(nwindFileName)
            End If
        End Sub



        Private Sub dashboardViewer1_DashboardLoaded(sender As Object, e As DashboardLoadedEventArgs) Handles dashboardViewer1.DashboardLoaded
            Dim viewer As DashboardViewer = CType(sender, DashboardViewer)
            UpdateDashboard(viewer.Dashboard)
        End Sub
    End Class
End Namespace

File: CS/WinFormsProject/ViewerForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.DashboardWin;
using DevExpress.DashboardCommon;
using DevExpress.DataAccess.ConnectionParameters;

namespace WinFormsProject
{
    public partial class ViewerForm : DevExpress.XtraEditors.XtraForm
    {
        public ViewerForm()
        {
            InitializeComponent();
        }
        public ViewerForm(string fileName) : this()
        {
            dashboardViewer1.LoadDashboard(fileName);
        }

        string nwindFileName = @"../../../nwind.mdb";
        private void dashboardViewer1_ConfigureDataConnection(object sender, DevExpress.DataAccess.ConfigureDataConnectionEventArgs e)
        {
            switch (e.ConnectionName)
            {
                case "nwindConnection":
                    Access97ConnectionParameters access = e.ConnectionParameters as Access97ConnectionParameters;
                    if (access != null)
                        access.FileName = nwindFileName;
                    break;
            }
        }

        private void dashboardViewer1_DashboardLoaded(object sender, EventArgs e)
        {
            DashboardViewer viewer = (DashboardViewer)sender;
            UpdateDashboard(viewer.Dashboard);           
        }       

        private void UpdateDashboard(Dashboard dashboard)
        {
            dashboard.DataSources.Clear();
            DataSource invoiceDataSource = new DataSource("RuntimeInvoices");
            dashboard.DataSources.Add(invoiceDataSource);
            foreach (var item in dashboard.Items.Where(i => i as DataDashboardItem != null).Cast<DataDashboardItem>())
            {
                item.DataSource = invoiceDataSource;
            }
        }

        private void dashboardViewer1_DataLoading(object sender, DevExpress.DataAccess.DataLoadingEventArgs e)
        {
            if (e.DataSourceName == "RuntimeInvoices")
                e.Data = NwindDataAdapter.NwindHelper.GetInvoices(nwindFileName);                         
        }


     
    }
}